sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -211,30 +246,33 @@ public void finish() {
     this.finish(this.getStatus());
   }
 
+  @SuppressWarnings({"JdkObsolete", "JavaUtilDate"})
   @Override
   public void finish(@Nullable SpanStatus status) {
+    cancelTimer();
```

Comment:
note - we won't re-schedule the timer when a new span gets added, but rather continuously wait until the last added span finishes, I kinda followed dart implementation here https://github.com/getsentry/sentry-dart/blob/28c88c966e2996591ea4363342080d9a003ad072/dart/lib/src/sentry_tracer.dart#L56-L58. Dunno if that's what we want exactly?

Reply:
Good question, let's discuss the trade-offs today.

Reply:
Discussed [here](https://www.notion.so/sentry/UI-events-Transactions-26057935c07c43c9981f2f62ac48780c) and agreed to keep it like in the dart SDK.